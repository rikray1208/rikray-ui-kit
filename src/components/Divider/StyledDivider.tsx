import React from 'react';
import styled, { css } from 'styled-components';

import { setFontSize } from '../../utils/helpers';

import type { DividerProps } from './index';
export const DividerWrapper = styled.div<Required<Pick<DividerProps, 'vertical'>>>`
    ${({ theme, vertical }) => css`
        ${setFontSize('Large', 'Divider', {})};
        ${vertical
            ? css`
                  display: inline;
                  border-inline-start: ${theme.Divider.borderWidth} ${theme.Divider.borderStyle}
                      ${theme.Divider.colorBorderBase};
                  vertical-align: middle;
                  margin: 0 ${theme.Divider['margin-1'] + 'rem'};
                  background: #000;
              `
            : css`
                  display: flex;
                  align-items: center;
                  height: fit-content;
                  width: 100%;
              `}
    `}
`;
export const Line = styled.div<Required<Pick<DividerProps, 'dashed' | 'titlePosition' | 'sideGap'>>>`
    ${({ titlePosition, sideGap, dashed, theme }) => css`
        height: 1px;
        width: 100%;

        &:first-child {
            width: ${titlePosition == 'Left' && sideGap + 'px'};
        }
        &:last-child {
            width: ${titlePosition == 'Right' && sideGap + 'px'};
        }
        border: ${theme.Divider.borderWidth} ${dashed ? 'dashed' : theme.Divider.borderStyle}
            ${theme.Divider.colorBorderBase};
    `};
`;

export const TextContainer = styled.span<Required<Pick<DividerProps, 'titlePosition' | 'space'>>>`
    ${({ titlePosition, space }) => css`
        padding: 0 ${space + 'px'};
        ${titlePosition == 'Left' &&
        css`
            margin-right: auto;
        `}
        ${titlePosition == 'Right' &&
        css`
            margin-left: auto;
        `}
    `}
`;
