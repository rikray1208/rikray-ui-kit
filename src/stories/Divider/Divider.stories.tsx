import { ComponentMeta, ComponentStory } from '@storybook/react';
import React, { FC } from 'react';

import { Divider, ThemeProvider } from '../../index';

const decorators = [
    (Story: any) => (
        <ThemeProvider>
            <>
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Architecto consectetur culpa dicta, dolores
                expedita fuga nobis odio quae quam voluptatibus! Dolorem earum esse fugiat minus natus qui soluta
                tempora, temporibus.
                <Story />
                Lorem ipsum dolor sit amet, consectetur adipisicing elit. Consequuntur, optio!
            </>
        </ThemeProvider>
    ),
];

export default {
    title: 'Design System/Global/Divider',
    component: Divider,
    decorators,
    argTypes: {
        titlePosition: {
            options: ['Left', 'Center', 'Right'],
            description: 'Меняет позицию загаловка',
            control: { type: 'radio' },
        },
        space: {
            description: 'Отступы от загаловка',
            control: { type: 'boolean' },
        },
        sideGap: {
            description: 'Отступы линий divider от края стороны',
            control: { type: 'number' },
        },
        dashed: {
            description: 'задает тип линий divider',
            control: { type: 'boolean' },
        },
        vertical: {
            description: 'Позволяет изменить направление на вертикальное',
            control: { type: 'boolean' },
        },
    },
} as ComponentMeta<typeof Divider>;

const Template: ComponentStory<typeof Divider> = (args) => <Divider {...args} />;

export const DefaultDivider = Template.bind({});

DefaultDivider.args = {
    children: 'Text',
    titlePosition: 'Center',
};
