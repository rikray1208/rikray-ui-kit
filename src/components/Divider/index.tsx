import React, { FC } from 'react';

import { DividerWrapper, Line, TextContainer } from './StyledDivider';

export const Positions = {
    Left: 'Left',
    Center: 'Center',
    Right: 'Right',
} as const;

export type DividerPositions = (typeof Positions)[keyof typeof Positions];

export interface DividerProps {
    titlePosition?: DividerPositions;
    vertical?: boolean;
    dashed?: boolean;
    space?: number;
    sideGap?: number;
    children?: React.ReactNode;
}

const Divider: FC<DividerProps> = ({ children, ...props }) => {
    const { titlePosition = 'Center', dashed = false, vertical = false, space = 10, sideGap = 20 } = props;
    return (
        <DividerWrapper vertical={vertical}>
            {!vertical && (
                <>
                    <Line sideGap={sideGap} titlePosition={titlePosition} dashed={dashed} />
                    {children && (
                        <TextContainer titlePosition={titlePosition} space={space}>
                            {children}
                        </TextContainer>
                    )}
                    <Line sideGap={sideGap} titlePosition={titlePosition} dashed={dashed} />
                </>
            )}
        </DividerWrapper>
    );
};

export default Divider;
