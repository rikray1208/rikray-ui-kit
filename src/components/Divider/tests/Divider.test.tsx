import React from 'react';
import { create } from 'react-test-renderer';

import { ThemeProvider } from '../../../index';

import Divider from '../index';

describe('Divider', () => {
    it('should return a Divider', () => {
        const element = (
            <ThemeProvider>
                <Divider />
            </ThemeProvider>
        );
        expect(create(element).toJSON()).toMatchSnapshot();
    });
    it('should return a vertical Divider', () => {
        const element = (
            <ThemeProvider>
                <Divider vertical />
            </ThemeProvider>
        );
        expect(create(element).toJSON()).toMatchSnapshot();
    });
    it('should return a dashed Divider', () => {
        const element = (
            <ThemeProvider>
                <Divider dashed />
            </ThemeProvider>
        );
        expect(create(element).toJSON()).toMatchSnapshot();
    });
    it('should return a Divider with title', () => {
        const element = (
            <ThemeProvider>
                <Divider>Title</Divider>
            </ThemeProvider>
        );
        expect(create(element).toJSON()).toMatchSnapshot();
    });
    it('should return a Divider with left title', () => {
        const element = (
            <ThemeProvider>
                <Divider titlePosition={'Left'}>Title</Divider>
            </ThemeProvider>
        );
        expect(create(element).toJSON()).toMatchSnapshot();
    });
    it('should return a Divider with right title', () => {
        const element = (
            <ThemeProvider>
                <Divider titlePosition={'Right'}>Title</Divider>
            </ThemeProvider>
        );
        expect(create(element).toJSON()).toMatchSnapshot();
    });
});
