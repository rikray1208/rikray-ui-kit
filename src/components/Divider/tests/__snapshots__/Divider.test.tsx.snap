// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should return a Divider 1`] = `
<div
  className="sc-kMjNwy guHMzt"
>
  <div
    className="sc-ilhmMj fPBMgh"
  />
  <div
    className="sc-ilhmMj fPBMgh"
  />
</div>
`;

exports[`Divider should return a Divider with left title 1`] = `
<div
  className="sc-kMjNwy guHMzt"
>
  <div
    className="sc-ilhmMj fpcCpj"
  />
  <span
    className="sc-jIRcFI cafQor"
  >
    Title
  </span>
  <div
    className="sc-ilhmMj fpcCpj"
  />
</div>
`;

exports[`Divider should return a Divider with right title 1`] = `
<div
  className="sc-kMjNwy guHMzt"
>
  <div
    className="sc-ilhmMj csdLvb"
  />
  <span
    className="sc-jIRcFI erHcis"
  >
    Title
  </span>
  <div
    className="sc-ilhmMj csdLvb"
  />
</div>
`;

exports[`Divider should return a Divider with title 1`] = `
<div
  className="sc-kMjNwy guHMzt"
>
  <div
    className="sc-ilhmMj fPBMgh"
  />
  <span
    className="sc-jIRcFI DanrR"
  >
    Title
  </span>
  <div
    className="sc-ilhmMj fPBMgh"
  />
</div>
`;

exports[`Divider should return a dashed Divider 1`] = `
<div
  className="sc-kMjNwy guHMzt"
>
  <div
    className="sc-ilhmMj eJsRUv"
  />
  <div
    className="sc-ilhmMj eJsRUv"
  />
</div>
`;

exports[`Divider should return a vertical Divider 1`] = `
<div
  className="sc-kMjNwy fYTzZa"
/>
`;
